cli/src/semgrep/commands/ci.py

```diff
@@ -415,25 +415,35 @@ def ci(
     # Split up rules into respective categories:
     blocking_rules: List[Rule] = []
     nonblocking_rules: List[Rule] = []
+    prev_scan_rules: List[Rule] = []
     cai_rules: List[Rule] = []
     for rule in filtered_rules:
         if "r2c-internal-cai" in rule.id:
             cai_rules.append(rule)
+        elif rule.is_prev_scan:
+            prev_scan_rules.append(rule)
+        elif rule.is_blocking:
+            blocking_rules.append(rule)
         else:
-            if rule.is_blocking:
-                blocking_rules.append(rule)
-            else:
-                nonblocking_rules.append(rule)
+            nonblocking_rules.append(rule)
 
     # Split up matches into respective categories
     blocking_matches_by_rule: RuleMatchMap = defaultdict(list)
     nonblocking_matches_by_rule: RuleMatchMap = defaultdict(list)
     cai_matches_by_rule: RuleMatchMap = defaultdict(list)
 
+    # Remove the prev scan matches by the rules that are in the current scan
+    # Done before the next loop to avoid interfering with ignore logic
+    removed_prev_scan_matches = {
+        rule: [match for match in matches if (not match.is_prev_scan)]
```

Comment:
Do we need this? Doesn't the `if rule.is_prev_scan` two lines below this do this already?

Reply:
You're right haha, we dont't! It was just me being paranoid.